nixos/common/flake.nix

```
@@ -7,6 +7,21 @@
 
 let
   cfg = config.srvos;
+
+  flake-registry = pkgs.writers.writeJSON "flake-registry.json" {
+    version = 2;
+    flakes = pkgs.lib.mapAttrsToList (name: value: {
+      exact = true;
+      from = {
+        id = name;
+        type = "indirect";
+      };
+      to = {
+        path = toString value;
+        type = "path";
+      };
+    }) cfg.flake.inputs;
```

```suggestion
    }) ({ self = cfg.flake; } // cfg.flake.inputs);
```

Or something like this? Not sure about the name, maybe `origin`, `system` or just `flake` would be a better choice? 

But I think it's useful to have the flake itself if available, i.e. to explore the systems config in a nix repl. Not sure whether we would even need the inputs, except nixpkgs, separately anymore if we do this.

---

Sounds good. `self` also maps what you get in the output function.